lib/navigation/imu_displacement.cpp

```diff
@@ -0,0 +1,50 @@
+#include "imu_displacement.hpp"
+
+#include <cmath>
+
+#include <chrono>
+
+namespace hyped::navigation {
+
+ImuDisplacement::ImuDisplacement()
+    : imu_displacement_(0),
+      imu_velocity_(0),
+      previous_timestamp_micros_(std::chrono::duration_cast<std::chrono::microseconds>(
+                                   std::chrono::system_clock::now().time_since_epoch())
+                                   .count())
+{
+}
+
+void ImuDisplacement::updateImuDisplacement(const core::Float imu_acceleration)
+{
+  const core::TimePoint time_now = std::chrono::system_clock::now();
+  const std::int64_t time_now_micros
+    = std::chrono::duration_cast<std::chrono::microseconds>(time_now.time_since_epoch()).count();
```

Comment:
have a look at `core::ITimeSource` :)

don't make your code untestable!

Reply:
So make a class which inherits from `ITimeSource` and implement the functionality I want in there so that we can make a similar class that can be used to create tests but with the same basic structure?

Reply:
not quite. just pass have your functions take an instance of `core::ITimeSource`

then the caller can make it e.g. a `core::WallClock` (for production) or `utils::ManualTime` (for testing)